### PR TITLE
Dual Stack IPv4 and IPv6 support.

### PR DESCRIPTION
**Pull request by raidensnake**
This is a draft to add IPv6 compatibility to the client.

### CODE REVIEW

File: src/networkmanager.cpp [hunk 1]
```diff
@@ -57,8 +58,8 @@ void NetworkManager::connect_to_server(server_type p_server)
 {
   server_socket->close();
   server_socket->abort();
-
-  server_socket->connectToHost(p_server.ip, p_server.port);
+  QHostAddress addr(p_server.ip);
+  server_socket->connectToHost(addr, p_server.port);
```
Comment (oldmud0) [inline src/networkmanager.cpp:62]: This should not make any difference. See the source for [`void QAbstractSocket::connectToHost(const QHostAddress &address, quint16 port, OpenMode openMode)`](https://code.woboq.org/qt5/qtbase/src/network/socket/qabstractsocket.cpp.html#_ZN15QAbstractSocket13connectToHostERK12QHostAddresst6QFlagsIN9QIODevice12OpenModeFlagEE).

File: src/text_file_functions.cpp [hunk 2]
```diff
@@ -223,7 +223,7 @@ QVector<server_type> AOApplication::read_serverlist_txt()
   while (!in.atEnd()) {
     QString line = in.readLine();
     server_type f_server;
-    QStringList line_contents = line.split(":");
+    QStringList line_contents = line.split(",");
```
Comment (stonedDiscord) [inline src/text_file_functions.cpp:226]: this breaks all past favorite lists

File: src/networkmanager.cpp (hunk 1 above)
Comment (in1tiate) [reply src/networkmanager.cpp]: Correct, tested without this and it works either way. Removed as of `8932a30`

File: src/text_file_functions.cpp (hunk 2 above)
Comment (in1tiate) [reply src/text_file_functions.cpp]: Fixed in `8932a30`

File: src/text_file_functions.cpp [hunk 3]
```diff
@@ -223,7 +223,12 @@ QVector<server_type> AOApplication::read_serverlist_txt()
   while (!in.atEnd()) {
     QString line = in.readLine();
     server_type f_server;
-    QStringList line_contents = line.split(":");
+    QStringList line_contents;
+
+    if (!line.contains(','))
```
Comment (stonedDiscord) [inline src/text_file_functions.cpp:228]: nice

**Review by stonedDiscord — APPROVED** [review]

File: src/text_file_functions.cpp (hunk 3 above)
Comment (oldmud0) [reply src/text_file_functions.cpp:228]: What if the server name contains a comma?

Comment (scatterflower) [reply src/text_file_functions.cpp:228]: should use regex instead

Comment (scatterflower) [reply src/text_file_functions.cpp:228]: image
image

**Review by scatterflower — CHANGES_REQUESTED** [review]
This entire PR feels like adding more complexity and edge cases to what is already really weird file handling. This should be written in a way that either:
1. Supports the old serverlist format without needing a separate code path or depending on some heuristic check which can easily fail
2. Breaks compatibility with the old format, and converts peoples' files for them.

My personal belief is that serverlists.txt should be ignored if serverlists.json exists, and if it does, we can parse/store it all as a json (or maybe ini?)
This means:
No weird regex for getting out the details
Can put ipv4/ipv6 addresses for a single server in one json entry
Can add server descriptions very easily to the favs list
Maintains backwards compatibility if .json file doesn't exist
Well standardized format can be edited by other tools

**Comment by raidensnake** [comment]
The Ipv6 range needs to be between 0-81 since it would need to include the colons and the brackets.

**Comment by raidensnake** [comment]
If anyone wishes to contribute to this please feel free to.

**Pull request closed without merging**